hack/tools/go.mod

```diff
@@ -20,14 +20,14 @@ require (
 	k8s.io/apimachinery v0.27.2
 	k8s.io/client-go v0.27.2
 	k8s.io/klog/v2 v2.90.1
-	k8s.io/kube-openapi v0.0.0-20230501164219-8b0f38b5fd1f
-	k8s.io/utils v0.0.0-20230209194617-a36077c30491
+	k8s.io/kube-openapi v0.0.0-20230601164746-7562a1006961
```

Comment:
This doesn't compile because proto.NewOpenAPIData in now expects a Document from gnostic-models and not from gnostic
```
/home/sbuerin/go/pkg/mod/k8s.io/client-go@v0.27.2/applyconfigurations/meta/v1/unstructured.go:64:38: cannot use doc (variable of type *"github.com/google/gnostic/openapiv2".Document) as *"github.com/google/gnostic-models/openapiv2".Document value in argument to proto.NewOpenAPIData
```

Basically this bump breaks the compatiblity with client-go.

I would suggest to add sigs.k8s.io/kustomize/api to the bump exclude list